docs/core-concepts.md

```diff
@@ -0,0 +1,52 @@
+# Core Concepts
+
+This document provides a high level overview of the Core Concepts that are embedded in the `cdk-eks-blueprint` framework. For the purposes of this document, we will assume the reader is familiar with Git, Docker, Kubernetes and AWS. 
+
+| Concept       | Description                                                           |     
+|---------------|-----------------------------------------------------------------------|
+| [Blueprint](#blueprint) | A `blueprint` combines `clusters`, `add-ons`, and `teams` into a cohesive object that can deployed as a whole. |                             
+| [Cluster](#cluster) | A Well-Architected EKS Cluster. |
+| [Add-on](#add-on) | A tool or service, represented by Kubernetes or AWS resources, which supports Kubernetes workloads. |
```

Comment:
We could expand the definition of add-on. Here is the one taken from EKS managed add-on documentation and expanded for our case:

EKS add-ons allows you to configure, deploy, and update the operational software, or add-ons, that provide key functionality to support your Kubernetes applications. Add-ons include tools for cluster networking like Calico CNI support for network policies, as well as operational software for observability, management, scaling, and security.

Reply:
I see you expanded on it later on so might be something to consider for the expanded definition.

Reply:
I like that.